### PR TITLE
Added logger. see test for example

### LINKED ISSUE #25: Server: inject real logger instead of console (like winston.js)

### PR DESCRIPTION
**Pull request by azimmerman12**
closes #25

### CODE REVIEW

**Comment by ugatenio** [comment]
please run `npm run fixlint`

File: src/server/app.ts [hunk 1]
```diff
@@ -1,87 +1,112 @@
...
+} from "./controllers";
+import { ConductorMock } from "./conductor/conductor-client-mock";
+import {
+  getLogger,
+  ServerLogger,
+  InternalLogger
+} from "./logger/logger.factory";
 
-export class Server {
+interface t {}
```
Comment (ugatenio) [inline src/server/app.ts:28]: ?

File: src/server/logger/logger.factory.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        filename: "./logs/server-error-%DATE%.log",
+        datePattern: "DD-MM-YYYY",
+        level: "debug"
+      }),
+      new transports.Console({ level: "info" })
+    ]
+  });
+
+export const getLogger = (): Logger => {
+  if (!isLoggerInitialized) init();
```
Comment (ugatenio) [inline src/server/logger/logger.factory.ts:38]: where do you set it to true?

File: src/server/logger/logger.factory.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import * as dailyWinston from "winston-daily-rotate-file";
+
+const logOptions = format.combine(
+  format.colorize(),
+  format.timestamp(),
+  format.align(),
+  format.printf(entry => `${entry.timestamp} ${entry.level} ${entry.message}`)
+);
+
+const loggerName = "serverLogger";
```
Comment (ugatenio) [inline src/server/logger/logger.factory.ts:11]: we need log file per service

File: src/server/services/node.service.ts [hunk 4]
```diff
@@ -1,63 +1,78 @@
...
+import { NodeInfo } from "../models";
+import { Service, Container } from "typedi";
+import { IService } from ".";
+import { DataStore } from "../db";
+import { ServerLogger, InternalLogger } from "../logger/logger.factory";
 
 @Service()
 export class NodeService implements IService {
+  private readonly dataStore = new DataStore<NodeInfo>();
+  private readonly logger: InternalLogger = Container.get(ServerLogger);
```
Comment (ugatenio) [inline src/server/services/node.service.ts:10]: user @Inject

File: src/server/services/node.service.ts [hunk 5]
```diff
@@ -1,63 +1,78 @@
...
-                            Object.assign(new NodeInfo(), this.dataStore.get(candidate.hostname), candidate) || candidate;
+    return this.dataStore.get(hostname);
+  }
 
-        await nodeInfo.save();
-        this.dataStore.set(nodeInfo.hostname, nodeInfo);
-        return nodeInfo;
-    }
+  async create(candidate: NodeInfo): Promise<NodeInfo> {
+    this.logger.info(`candidate: ${candidate.cluster_name} is created`);
```
Comment (ugatenio) [inline src/server/services/node.service.ts:42]: move it after creation

File: src/server/app.ts [hunk 6]
```diff
@@ -1,87 +1,112 @@
...
-        await this.server.listen(this.port, () => {});
-    }
+  async run(): Promise<void> {
+    Container.set(Database, (this.production && new Redis()) || new DbMock());
+    Container.set(SocketIO, this.io);
+    Container.set(
+      Conductor,
+      (this.production && new Conductor()) || new ConductorMock()
+    );
+    Container.set(ServerLogger, getLogger());
```
Comment (ugatenio) [inline src/server/app.ts:94]: use production flag. if not it production just print to console

File: src/server/app.ts (hunk 1 above)
Comment (azimmerman12) [reply src/server/app.ts:28]: ;)

File: src/server/logger/logger.factory.ts (hunk 2 above)
Comment (azimmerman12) [reply src/server/logger/logger.factory.ts:38]: done

File: src/server/services/node.service.ts (hunk 4 above)
Comment (azimmerman12) [reply src/server/services/node.service.ts:10]: doesnt work in this case

File: src/server/logger/logger.factory.ts (hunk 3 above)
Comment (azimmerman12) [reply src/server/logger/logger.factory.ts:11]: why? why not log per process? do you want to cross files to see what happened?

File: src/server/services/node.service.ts (hunk 5 above)
Comment (azimmerman12) [reply src/server/services/node.service.ts:42]: this was just a sample. removing

File: src/server/app.ts (hunk 6 above)
Comment (azimmerman12) [reply src/server/app.ts:94]: sababa

File: src/server/services/node.service.ts (hunk 4 above)
Comment (ugatenio) [reply src/server/services/node.service.ts:10]: use class wrapper and it will

Comment (azimmerman12) [reply src/server/services/node.service.ts:10]: i know but see why. ok will wrap

**Comment by ugatenio** [comment]
please replace all console with the new logger

File: src/server/logger/logger.factory.ts (hunk 3 above)
Comment (ugatenio) [reply src/server/logger/logger.factory.ts:11]: you right but I want the ability to do it. for example for conductor client

Comment (azimmerman12) [reply src/server/logger/logger.factory.ts:11]: still under the same process. i will add the ability for strongly suggest to write to the same log file so we won't need to cross between files.

**Pull request merged**